frontend/src/contexts/AppContext.tsx

```diff
@@ -18,25 +25,59 @@ interface State {
   conferenceStatus: IConferenceStatus;
 }
 
-const reducer = (state, action) => {
-  const allowedActions = [
-    'FISHBOWL_STARTED',
-    'FISHBOWL_READY',
-    'FISHBOWL_STATUS',
-    'FISHBOWL_ENDED',
-    'JOIN_GUEST',
-    'JOIN_USER'
-  ];
-  const { type, ...actionData } = action;
+type fishbowlStartedAction = {
+  type: typeof FISHBOWL_STARTED;
+  fishbowlStarted: boolean;
+};
+
+type fishbowlReadyAction = {
+  type: typeof FISHBOWL_READY;
+  fishbowlReady: boolean;
+};
+
+type fishbowlStatusAction = {
+  type: typeof FISHBOWL_STATUS;
+  fishbowlReady?: boolean;
+  fishbowlStarted?: boolean;
+  isGuest?: boolean;
+  prejoin?: boolean;
+  conferenceStatus: IConferenceStatus;
```

Comment:
We are leaking the details of how to reset a fishbowl. This should only expose actions and needed payload, but not the full data of the state. Otherwise the responsible of changing the state is not the reducer itself, but the entire application.